GVFS/GVFS.Common/Maintenance/GitMaintenanceStep.cs

```diff
@@ -276,6 +276,9 @@ protected void LogErrorAndRewriteCommitGraph(ITracer activity, List<string> pack
             string commitGraphPath = Path.Combine(this.Context.Enlistment.GitObjectsRoot, "info", "commit-graph");
             errorMetadata["TryDeleteFileResult"] = this.Context.FileSystem.TryDeleteFile(commitGraphPath);
 
+            string commitGraphsPath = Path.Combine(this.Context.Enlistment.GitObjectsRoot, "info", "commit-graphs");
+            errorMetadata["TryDeleteDirectoryResult"] = this.Context.FileSystem.TryDeleteDirectory(commitGraphsPath, out Exception _);

```

Comment:
Would it be useful to include the `Exception` (if present) in `errorMetadata` as well?